### PR TITLE
MGMT-7730: Add Bound condition to Agent CRD

### PR DESCRIPTION
**Pull request by rollandf**
# Assisted Pull Request

## Description
Add Bound condition to Agent CRD

Signed-off-by: Fred Rolland <frolland@redhat.com>

## List all the issues related to this PR

- [x] New Feature
- [ ] Bug fix
- [ ] Tests
- [ ] Documentation
- [ ] CI/CD 

## What environments does this code impact?

- [ ] Automation (CI, tools, etc)
- [ ] Cloud
- [x] Operator Managed Deployments
- [ ] None

## How was this code tested?

- [ ] assisted-test-infra environment
- [ ] dev-scripts environment
- [ ] Reviewer's test appreciated
- [x] Waiting for CI to do a full test run
- [ ] Manual (Elaborate on how it was tested)
- [ ] No tests needed

## Assignees

/cc @nmagnezi 
/cc @filanov 
/cc @yevgeny-shnaidman 
/cc @avishayt 

## Checklist

- [x] Title and description added to both, commit and PR.
- [x] Relevant issues have been associated (see [CONTRIBUTING] guide)
- [x] Reviewers have been listed
- [x] This change does not require a documentation update (docstring, `docs`, README, etc)
- [x] Does this change include unit-tests (note that code changes require unit-tests)

## Reviewers Checklist

- [ ] Are the title and description (in both PR and commit) meaningful and clear?
- [ ] Is there a bug required (and linked) for this change?
- [ ] Should this PR be backported?

[Kubernetes community documentation]: https://github.com/kubernetes/community/blob/master/contributors/guide/pull-requests.md#commit-message-guidelines
[CONTRIBUTING]: https://github.com/openshift/assisted-service/blob/master/CONTRIBUTING.md

### CODE REVIEW

**Comment by rollandf** [comment]
/hold
Should be rebased on top of #2477

File: internal/controller/controllers/agent_controller.go [hunk 1]
```diff
@@ -590,6 +621,37 @@ func requirementsMet(agent *aiv1beta1.Agent, status string) {
...
+		condStatus = corev1.ConditionFalse
+		reason = aiv1beta1.BindingReason
+		msg = aiv1beta1.BindingMsg
+	case models.HostStatusUnbinding:
+		condStatus = corev1.ConditionFalse
+		reason = aiv1beta1.UnbindingReason
+		msg = aiv1beta1.UnbindingMsg
+	case models.HostStatusDisconnectedUnbound, models.HostStatusKnownUnbound, models.HostStatusInsufficientUnbound,
+		models.HostStatusDisabledUnbound, models.HostStatusDiscoveringUnbound:
+		condStatus = corev1.ConditionTrue
```
Comment (yevgeny-shnaidman) [inline internal/controller/controllers/agent_controller.go:639]: i think it is supposed to be ConditionFalse, no?

Comment (rollandf) [reply internal/controller/controllers/agent_controller.go]: Yep. Fixed and added tests.

**Pull request merged**